pandas/core/frame.py

```diff
@@ -2903,8 +2903,76 @@ def to_html(
-        """
+
+        Notes
+        -----
+        As of version 1.4.0, a warning is shown for the following use of keyword
+        arguments if they are non-default values, in order to align this function with
+        the formatting structure akin to ``Styler.to_html``.
+
+          - ``classes``, likely replaced by ``table_attributes``.
+          - ``sparsify``, likely separated to ``sparse_index``, ``sparse_columns``.
+          - ``max_cols``, likely replaced with ``max_columns`` consistently.
+          - ``formatters``, likely replaced with ``formatter``.
+          - ``float_format``, likely replaced by ``precision``, ``decimal``,
+            ``thousands``.
+          - ``border``, likely removed due to deprecated HTML specification.
+          - ``col_space``, likely removed in favour of CSS solutions.
+          - ``justify``, likely removed in favour of CSS solutions.
+          - ``render_links``, likely removed due to limited functionality.
+          - ``notebook``, likely removed due to legacy impact.
+          - ``show_dimensions``, likely removed in favour of caption solution.
+          - ``bold_rows``, likely replaced with ``bold_headers`` with index and column
+            control.
+        """
+        # Warnings are shown in 1.4.0 in preparation for signature changes in 2.0.0
+        warnings_default_none = {
+            "classes": "replaced by `table_attributes`",
+            "sparsify": "replaced by `sparse_index` and `sparse_columns`",
+            "max_cols": "replaced by `max_columns` for consistency",
+            "formatters": "replaced by `formatter` accepted by `Styler.format`",
+            "float_format": "replaced by `precision`, `decimal`, and `thousands`",
+            "border": "removed as deprecated HTML, suggested to use CSS",
+            "col_space": "removed, suggested to use CSS `min-width: 100px;`",
+            "justify": "removed, suggested to use CSS",
+        }
+
+        warnings_default_false = {
+            "render_links": "removed due to limited functionality",
+            "notebook": "removed as a legacy argument",
+            "show_dimensions": "removed, suggested to use `caption=f'{df.shape}'`",
+        }
+
+        warnings_default_true = {
+            "bold_rows": "replaced by `bold_headers` controlling index and columns",
+        }
+
+        warning_msg = (
+            "In future versions `DataFrame.to_html` is expected to utilise the base "
+            "implementation of `Styler.to_html` for formatting and rendering. "
+            "The arguments signature may therefore change. You are specifically using "
+            "the following arguments: "
+        )
+        warning_flag = False
+
+        for kwarg, msg in warnings_default_none.items():
+            if locals()[kwarg] is not None:
+                warning_flag = True
+                warning_msg += f"\n    `{kwarg}`, which may be {msg}."
+        for kwarg, msg in warnings_default_false.items():
```

Comment:
For those cases where the default is not None (but True or False), we will also need to raise a deprecation warning if the user specified the keyword explicitly but with the default (as that will also stop working if the keyword is removed in the future). 
Typically, to solve this we change the default value in the signature to None and then if it is None, change it to True/False (depending on the actual default).